### DISCUSSION TITLE
What are the best practices for all flash cluster?

### DISCUSSION THREAD

**Discussion by microyahoo** [rook/rook#13133] [Q&A]
Does rook support spdk to maximize the performance of all flash? What are the best practices for all flash cluster? For example, how to deploy and what features need to be enabled? Thanks in advance. @travisn @satoru-takeuchi

**Comment by travisn** [comment]
This is the Crimson project in Ceph. It's not ready for production yet, but can be tested if desired. See the [Ceph Crimson](https://docs.ceph.com/en/quincy/dev/crimson/crimson/) topic, though we don't have specific instructions for Rook yet.

**Reply by 0xAlcibiades** [reply]
So, it doesn't seem like this needs to wait for crimson though, the bluestore backend supports it as a driver in reef from a read of the docs: https://docs.ceph.com/en/reef/rados/configuration/bluestore-config-ref/#spdk-usage

I'm hitting this in #13483 where I have some NVMEs that should be able to do 1m iops each, but the OSDs only get about 30k iops. Since it's a lab, I'm gonna take SPDK for a spin and report back.
